### PR TITLE
[WIP] Create a new component for the messages, remove useless var

### PR DESCRIPTION
**Pull request by lcordier42**
I simplified the chat a lot(i added a new component for messages), it does the same thing as before (apart from the functionality of connected members that I removed, it wasn't fully fonctionnal ...), and it is especially much more readable, if it's good for you I continue to from this base which is much easier to understand, from my point of view anyway.

### CODE REVIEW

File: components/Messages.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        await this.props.channel.on("messageAdded", this.messageAdded);
+    }
+
+    public componentWillUnmount() {
+        this.props.channel.removeListener("messageAdded", this.messageAdded);
+    }
+
+    public async componentWillReceiveProps(nextProps: any) {
+        // If we change channel nextProps !== this.props so we have to load messages for the new channel
+        if (nextProps !== this.props) {
```
Comment (brikou) [inline components/Messages.tsx]: I don't think it works as you are comparing 2 objects, even if looks the same, they are always different, for example:

```js
console.log({foo:true}==={foo:true}) // false
```

Comment (lcordier42) [reply components/Messages.tsx]: Ok, i'll change it

**Comment by brikou** [comment]
:+1:

File: components/Channels.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import * as React from "react";
+
+export class Channels extends React.Component<{
+    channels: any[];
+    // what is the type of a function?
```
Comment (brikou) [inline components/Channels.tsx:5]: keep the comment as is, I'll update type later

File: components/Messages.tsx [hunk 3]
```diff
@@ -39,6 +41,8 @@ export class Messages extends React.Component<
             const messagePage = await nextProps.channel.getMessages();
             this.setState({ messages: messagePage.items });
             await nextProps.channel.on("messageAdded", this.messageAdded);
+            // Added to save the current channel and stay on after refresh page
+            document.cookie = this.props.channel.uniqueName;
```
Comment (brikou) [inline components/Messages.tsx]: use localStorage instead

**Comment by brikou** [comment]
great until now

**Pull request merged**